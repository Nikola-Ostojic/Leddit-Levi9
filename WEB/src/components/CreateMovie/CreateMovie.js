import React, { useState } from "react";
import { Button, TextField } from "@material-ui/core";
import { toast } from "react-toastify";

import { apiCall } from "utils/ApiCall";

const CreateMovie = ({ loadMovies }) => {
  const [name, setName] = useState("");
  const [imageUrl, setImageUrl] = useState("");

  const createMovie = (e) => {
    e.preventDefault();
    apiCall({
      url: "/movies",
      method: "POST",
      data: { name, imageUrl },
      requiresAuth: true
    })
      .then(() => {
        toast.success(`Movie: ${name} created!`);
        // Cleaning up the state after a successful creation
        setName("");
        setImageUrl("");
      })
      .catch((e) => {
        toast.error("Failed to create movie");
      })
      .finally(loadMovies);
  };

  return (
    <form onSubmit={createMovie}>
      <TextField
        fullWidth
        margin="normal"
        label="Movie name"
        type="text"
        required
        onChange={({ target: { value } }) => setName(value)}
        value={name}
      />
      <TextField
        fullWidth
        margin="normal"
        label="Image url"
        type="text"
        onChange={({ target: { value } }) => setImageUrl(value)}
        value={imageUrl}
      />
      <Button variant="contained" color="primary" type="submit">
        Create movie
      </Button>
    </form>
  );
};

export default CreateMovie;
