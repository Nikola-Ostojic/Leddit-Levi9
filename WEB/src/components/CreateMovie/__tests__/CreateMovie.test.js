import React from "react";
import { shallow } from "enzyme";

import CreateMovie from "../CreateMovie";

describe("CreateMovie component tests", () => {
  it("Should match snapshot", () => {
    const wrapper = shallow(<CreateMovie />);

    expect(wrapper).toMatchSnapshot();
  });
});
