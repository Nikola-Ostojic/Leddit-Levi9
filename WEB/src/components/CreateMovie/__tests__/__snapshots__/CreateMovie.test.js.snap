// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateMovie component tests Should match snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <WithStyles(ForwardRef(TextField))
    fullWidth={true}
    label="Movie name"
    margin="normal"
    onChange={[Function]}
    required={true}
    type="text"
    value=""
  />
  <WithStyles(ForwardRef(TextField))
    fullWidth={true}
    label="Image url"
    margin="normal"
    onChange={[Function]}
    type="text"
    value=""
  />
  <WithStyles(ForwardRef(Button))
    color="primary"
    type="submit"
    variant="contained"
  >
    Create movie
  </WithStyles(ForwardRef(Button))>
</form>
`;
